### PR TITLE
Added QR Generation page on mobile

### PR DESCRIPTION
**Pull request by firtap**

### CODE REVIEW

File: ScanShop.Mobile/ScanShop.Mobile/ViewModels/QRGenerateViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+using System.Collections.Generic;
+using System.Text;
+using Xamarin.Essentials;
+using Xamarin.Forms;
+
+namespace ScanShop.Mobile.ViewModels
+{
+    public class QRGenerateViewModel : BaseViewModel
+    {
+        private string _QRGenValue = "Hello World!";
```
Comment (Losdrew) [inline ScanShop.Mobile/ScanShop.Mobile/ViewModels/QRGenerateViewModel.cs:12]: I think you should move assignments for fields to the constructor

File: ScanShop.Mobile/ScanShop.Mobile/Views/QRScanPage.xaml [hunk 2]
```diff
@@ -14,11 +14,11 @@
         <Label Text="{Binding QRScanText}" />
         <Grid VerticalOptions="FillAndExpand" HorizontalOptions="FillAndExpand">
             <zxing:ZXingScannerView IsScanning="True" IsAnalyzing="True" 
-                VerticalOptions="FillAndExpand"
-                HorizontalOptions="FillAndExpand" 
-                ScanResultCommand="{Binding ScanCommand}"/>
+                    VerticalOptions="FillAndExpand"
```
Comment (Losdrew) [inline ScanShop.Mobile/ScanShop.Mobile/Views/QRScanPage.xaml:17]: Why did you put the tabs here?

File: ScanShop.Mobile/ScanShop.Mobile/Views/ProfilePage.xaml [hunk 3]
```diff
@@ -69,10 +69,10 @@
                     <Label Text="Discount:" Style="{StaticResource FrameLabel}" />
                 </Frame>
                 <Label Grid.Row="4" Grid.Column="1" Text="{Binding Path=User.Discount, StringFormat='{0}%'}"
-                       TextColor="#FD5A5A" Style="{StaticResource BlackLabel}"/>
+                        TextColor="#FD5A5A" Style="{StaticResource BlackLabel}"/>
```
Comment (Losdrew) [inline ScanShop.Mobile/ScanShop.Mobile/Views/ProfilePage.xaml:72]: You can also fix this after removing the tabs in QRScanPage.xaml

File: ScanShop.Mobile/ScanShop.Mobile/Views/QRScanPage.xaml (hunk 2 above)
Comment (firtap) [reply ScanShop.Mobile/ScanShop.Mobile/Views/QRScanPage.xaml]: I don't know.
Fixed.

File: ScanShop.Mobile/ScanShop.Mobile/Views/ProfilePage.xaml (hunk 3 above)
Comment (firtap) [reply ScanShop.Mobile/ScanShop.Mobile/Views/ProfilePage.xaml]: Fixed.

File: ScanShop.Mobile/ScanShop.Mobile/ViewModels/QRGenerateViewModel.cs [hunk 4]
```diff
@@ -9,15 +9,17 @@ namespace ScanShop.Mobile.ViewModels
...
+        private int _QRSize;
         private const float QRSizePercentage = 80;
 
         public Command GoBackCommand { get; }
 
         public QRGenerateViewModel()
         {
             GoBackCommand = new Command(OnGoBackClicked);
+            _QRGenValue = "Hello World!";
+            _QRSize = Convert.ToInt32(Math.Round(DeviceDisplay.MainDisplayInfo.Width / DeviceDisplay.MainDisplayInfo.Density * QRSizePercentage / 100));
```
Comment (Losdrew) [inline ScanShop.Mobile/ScanShop.Mobile/ViewModels/QRGenerateViewModel.cs:22]: This line is too long, better to split the results of mathematical operations into local variables

Comment (firtap) [reply ScanShop.Mobile/ScanShop.Mobile/ViewModels/QRGenerateViewModel.cs:22]: I don't think this is needed, it is only 3 arithmetic operations and we will not use these local variables elsewhere.

Comment (firtap) [reply ScanShop.Mobile/ScanShop.Mobile/ViewModels/QRGenerateViewModel.cs]: You are right.
Fixed.

**Review by Losdrew — APPROVED** [review]

**Pull request merged**